app/controllers/mixins/ems_common_angular.rb

```diff
@@ -713,8 +713,9 @@ def build_credentials(ems, mode)
     end
 
     def retrieve_event_stream_selection
-      return "amqp" if @ems.connection_configurations.ceilometer.try(:endpoint).nil? && @ems.connection_configurations.amqp.try(:endpoint)
-      "ceilometer"
+      return 'amqp' if @ems.connection_configurations.amqp&.endpoint&.hostname&.present?
+      return 'ceilometer' if @ems.connection_configurations.ceilometer&.endpoint&.hostname&.present?
+      'none'
```

Comment:
Noticed that a `hostname&.present?` condition has been added which did not exist before.
Should not matter for OpenStack Cloud/Infra Provider I think (where this method is used), but just wanted to know if it's absolutely needed for Nuage?

Reply:
Could you post a screenshot of the Nuage Edit Network Provider page?
Does it have AMQP/Ceilometer Radio buttons too?

Reply:
I think the above change may affect how we display the Radio buttons for OpenStack.

OpenStack selects the Ceilometer radiobutton by default if the AMQP endpoint does not exist.
But with the above change, there is an explicit check for the Ceilometer endpoint.
If both AMQP and Ceilometer endpoints do not exist, none of the radiobuttons would be selected -- which is not what we want.

Let's keep the OpenStack UI unaffected by this change.